### PR TITLE
Use web3-react to connect web3 provider

### PR DESCRIPTION
**Pull request by toshiakitakase**
Allow connecting to Metamask using web3-react.

### CODE REVIEW

**Comment by wild-mouse** [comment]
Could you resolve conflicts?

File: contract/.env.example [hunk 1]
```diff
@@ -1,2 +0,0 @@
-RPC_URL=
-MNEMONIC=
```
Comment (wild-mouse) [inline contract/.env.example:2]: This file is necessary to use `HDWalletProvider` for truffle and therefore to deploy contracts.
Could you restore it?

**Comment by toshiakitakase** [comment]
DONE

**Review by wild-mouse — APPROVED** [review]
LGTM 🏃

**Comment by wild-mouse** [comment]
Sorry left one comment.

**Review by toshiakitakase — COMMENTED** [review]
Mistake!w
But this MNEMONIC key is demo key.

**Comment by toshiakitakase** [comment]
DONE

**Review by wild-mouse — APPROVED** [review]
LGTM 🏃‍♀️

**Pull request merged**